cmd/geth/passwords.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	cli "gopkg.in/urfave/cli.v1"
+
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	vaultAPI "github.com/hashicorp/vault/api"
+	awsauth "github.com/hashicorp/vault/builtin/credential/aws"
+)
+
+func fetchPasswordFromVault(ctx *cli.Context) (string, error) {
+	if usingVaultPassword(ctx) {
+		// Authenticate to Vault via the AWS method
+		vaultConfig := vaultAPI.DefaultConfig()
```

Comment:
defaults to 60s time out and 3 retries for client make sure this is good cross region @Lsquared13

Reply:
Maybe do a cross-region test putting the nodes in one region and the primary-region in a region halfway around the world (e.g. us-east-1 and ap-southeast-2)>

Reply:
Cool, I'll run a private tx test w/ those settings now